File: app/src/main/java/seigneur/gauvain/mycourt/utils/ViewUtils.java
package seigneur.gauvain.mycourt.utils;

import android.graphics.Outline;
import android.view.View;
import android.view.ViewOutlineProvider;

public class ViewUtils {
    private ViewUtils() { }

    public static final ViewOutlineProvider CIRCULAR_OUTLINE = new ViewOutlineProvider() {
        @Override
        public void getOutline(View view, Outline outline) {
            outline.setOval(view.getPaddingLeft(),
                    view.getPaddingTop(),
                    view.getWidth() - view.getPaddingRight(),
                    view.getHeight() - view.getPaddingBottom());
        }
    };
}

File: app/src/main/java/seigneur/gauvain/mycourt/utils/ViewUtils.kt
package seigneur.gauvain.mycourt.utils

import android.graphics.Outline
import android.view.View
import android.view.ViewOutlineProvider

object ViewUtils {

    //need to cast ViewOutlineProvider() as object in kotlin because we want to slight modification of
    // it, without explicitly declaring a new subclass for (override fun here)
    val CIRCULAR_OUTLINE: ViewOutlineProvider = object : ViewOutlineProvider() {
        override fun getOutline(view: View, outline: Outline) {
            outline.setOval(view.paddingLeft,
                    view.paddingTop,
                    view.width - view.paddingRight,
                    view.height - view.paddingBottom)
        }
    }
}
